for v0.3: restructure, use click, allow more tests

purarue opened the pull request:
Still some features to implement/clean up

separate keepassxc-cli calls into wrapper module
use click instead of docopt #9
remove 3.7 dependent code to allow compatability
down to python3.5
cache repeated passwords to avoid unnecessary API calls

purarue committed:
for v0.3: major rewrite

Still some features to implement/clean up

separate keepassxc-cli calls into wrapper module
use click instead of docopt #9
remove 3.7 dependent code to allow compatability
down to python3.5
cache repeated passwords to avoid unnecessary API calls

seanbreckenridge commented:
Todo:

- [ ] Make logging respect quiet/verbose, add more verbose statements
- [ ] Finish adding typing; add mypy hook to pytest/travis
- [ ] raise an exception on >= 400 HTTP requests
- [ ] rewrite CLI main code

purarue committed:
v0.3 main features done

seperated test requirements into requirements-dev.txt
add 3.6 support
finish typing/add mypy test to travis
add exception for HTTP >= 400 errors
cache sha1 values to avoid multiple HTTP requests
for the same password
rewrite main to be more readable
added more comments/docstrings

purarue committed:
catch PwnedPasswordException, add more debug logs

purarue committed:
Merge README changes into v0.3

purarue committed:
allow user to provide password non interactively

using the KEEPASSXC_PWNED_PASSWD environment variable

purarue committed:
see if python 3.5 works

purarue committed:
remove 3.5 from travis

no support for variable annotation syntax

purarue committed:
basic authentication/count entry tests

purarue committed:
simulate environment variables

purarue committed:
add more tests

purarue committed:
ignore entries with no password

purarue committed:
update python classifiers and --help message

purarue committed:
fix mypy error: "Credential" has no attr password

purarue committed:
Retry on undocumented/429 request errors

purarue committed:
remove summary seperator
